backend/hitas/models/housing_company.py

```diff
@@ -95,6 +114,29 @@ def total_shares_count(self) -> int:
             .aggregate(sum_shares_count=Sum("shares_count"))["sum_shares_count"]
         )
 
+    @property
+    def over_thirty_years_old(self) -> bool:
+        from hitas.models import Apartment
+
+        completion_dates: list[datetime.date | None] = Apartment.objects.filter(
+            building__real_estate__housing_company=self
+        ).values_list("completion_date", flat=True)
+
+        if None in completion_dates:
+            return False
+
+        return relativedelta(timezone.now().date(), max(completion_dates)).years >= 30
```

Comment:
I would change the property name to signal the the return value is a boolean. Also, there shouldn't be a need to do the checking in python, as we can just ask the DB to return the needed value like so:

```py
    @property
    def is_over_thirty_years_old(self) -> bool:
        from hitas.models import Apartment

        completion_date: Optional[datetime.date] = (
            Apartment.objects.filter(building__real_estate__housing_company=self)
            .order_by("-completion_date")
            .first()
            .completion_date
        )

        if completion_date is None:
            return False

        return relativedelta(timezone.now().date(), completion_date).years >= 30
```

Reply:
Yes, I'll change this.